common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationSupport.java

```diff
@@ -37,9 +37,9 @@ public final class CollationSupport {
   public static class StringSplitSQL {
     public static UTF8String[] exec(final UTF8String s, final UTF8String d, final int collationId) {
       CollationFactory.Collation collation = CollationFactory.fetchCollation(collationId);
-      if (collation.supportsBinaryEquality) {
+      if (collation.isUtf8BinaryType) {
         return execBinary(s, d);
-      } else if (collation.supportsLowercaseEquality) {
+      } else if (collation.isUtf8LcaseType) {
```

Comment:
iiuc, this is just renaming currently

but how exactly do we plan to modify CollationSupport classes, so that we get the desired behaviour regarding space trimming integration?

Reply:
By incorporating `usesTrimCollation` flag we agreed on in the previous prs.

Reply:
so the idea is:
1. apply trimming policy to the arguments
2. proceed with `isUtf8BinaryType` / `isUtf8LcaseType` check (which is _less_ restrictive than the previously used `supportsBinaryEquality` / `supportsLowercaseEquality` check)

and this finally gives us the proper & desired behaviour for UTF8_BINARY_RTRIM

Reply:
Yes.

Reply:
sgtm